### PR TITLE
Deprecate replication classes, properties, etc.

### PR DESCRIPTION
**Pull request by dlmarion**

### CODE REVIEW

File: server/tserver/src/test/java/org/apache/accumulo/tserver/replication/AccumuloReplicaSystemTest.java [hunk 1]
```diff
@@ -55,16 +55,15 @@
...
 import org.apache.accumulo.tserver.logger.LogFileValue;
-import org.apache.accumulo.tserver.replication.AccumuloReplicaSystem.ReplicationStats;
-import org.apache.accumulo.tserver.replication.AccumuloReplicaSystem.WalClientExecReturn;
-import org.apache.accumulo.tserver.replication.AccumuloReplicaSystem.WalReplication;
 import org.apache.hadoop.fs.Path;
 import org.apache.hadoop.io.Text;
 import org.junit.Test;
 
+@SuppressWarnings("deprecation")
 public class AccumuloReplicaSystemTest {
```
Comment (ctubbsii) [inline server/tserver/src/test/java/org/apache/accumulo/tserver/replication/AccumuloReplicaSystemTest.java:63]: I think if you deprecate this test class, you don't need to suppress deprecations or deprecate individual methods.

File: core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java [hunk 2]
```diff
@@ -323,6 +323,7 @@ ConditionalWriter createConditionalWriter(String tableName, ConditionalWriterCon
    *
    * @return an object to modify replication configuration
    */
+  @Deprecated
```
[the remark this answers is not in the document]
Comment (dlmarion) [reply core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java]: Fixed in ef1b169

File: server/tserver/src/test/java/org/apache/accumulo/tserver/replication/AccumuloReplicaSystemTest.java (hunk 1 above)
Comment (dlmarion) [reply server/tserver/src/test/java/org/apache/accumulo/tserver/replication/AccumuloReplicaSystemTest.java]: Fixed in ef1b169

File: core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (ctubbsii) [reply core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java]: There's a few more places that could benefit from this. The properties, in particular, but also any interfaces that users might implement that are configurable in the properties.

[the remark this answers is not in the document]
Comment (dlmarion) [reply core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java]: I think I got them all.

**Review by ctubbsii — APPROVED** [review]

**Pull request merged**